src/main/java/com/spring/app/controllers/VacancyController.java

```diff
@@ -16,56 +18,45 @@
 import java.util.Locale;
 import java.util.Map;
 
+/**
+ * VacancyController
+ *
+ * @author lyubov
+ */
 @Controller
 public class VacancyController {
     @Autowired
     private VacancyRepository vacancyRepository;
 
     @GetMapping("vacancies")
-    public String main(Map<String, Object> model) {
-        Iterable<Vacancy> messages = vacancyRepository.findAll();
-
-        model.put("vacancies", messages);
-
-        return "vacancy";
+    public ResponseEntity<Iterable<Vacancy>> main(Map<String, Object> model) {
+        Iterable<Vacancy> vacancies = vacancyRepository.findAll();
+        return new ResponseEntity<>(vacancies, HttpStatus.OK);
     }
 
     @PostMapping("vacancies")
-    public String add(@RequestParam String name, @RequestParam String sSalary, @RequestParam String sDate, Map<String, Object> model) {
+    public ResponseEntity<Vacancy> add(@RequestParam String name, @RequestParam String sSalary, @RequestParam String sDate, Map<String, Object> model) {
         DateFormat df = new SimpleDateFormat("yyyy-mm-dd", Locale.ENGLISH);
         Date date = new Date();
         try {
             date = df.parse(sDate);
         } catch (ParseException e) {
             e.printStackTrace();
         }
-
         float salary = Float.parseFloat(sSalary);
-
-
         Vacancy vacancy = new Vacancy(name, salary, date);
             vacancyRepository.save(vacancy);
-
-
-        Iterable<Vacancy> vacancies = vacancyRepository.findAll();
-
-        model.put("vacancies", vacancies);
-
-        return "vacancy";
+        return new ResponseEntity<>(vacancy, HttpStatus.OK);
     }
 
-    @PostMapping("vacancies/filter")
-    public String filter(@RequestParam String filter, Map<String, Object> model) {
+    @GetMapping("vacancies/filter")
```

Comment:
code dublicate

Reply:
I've removed post method. I don't need it